lib/dagwood/dependency_graph.rb

```diff
@@ -24,6 +24,10 @@ def reverse_order
       @reverse_order ||= order.reverse
     end
 
+    def parallel_reverse_order
+      parallel_order.reverse
```

Comment:
This should work, but my graph theory knowledge isn't quite advanced enough to convince me that its optimal... I'm just thinking of something like 

B depends on A
C depends on A
D depends on C

In that case, I'd expect forwards to be [[A], [B,C], [D]]
But in reverse, couldn't we do [[B, D], [C], [A]]?

i.e. is optimal computed by "reverse the direction of the edge between each pair of nodes", and then recompute parallel_order on the resulting graph? I'm in over my head here... 

I think what you've got works though, and is probably fast enough and more readable than something optimal(assuming this isn't already optimal).

Reply:
I'd be curious to see what this implementation could look like. I think this is good enough for now though

Reply:
@Adverbly I agree. I think your example correctly points out that the simply reversing the `parallel_order` may not yield the same result as re-running the algorithm on the reversed dependencies themselves. I'm unsure at this time if this could result in errors (I can't think of any cases where it would be an issue), so until we actually encounter a problem with this, I'd say it's good enough.